x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java

```diff
@@ -41,198 +38,156 @@ public AuditTrailService(List<AuditTrail> auditTrails, XPackLicenseState license
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.authenticationFailed(requestId, realm, token, request);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.authenticationFailed(requestId, realm, token, request);
         }
     }
 
     @Override
     public void accessGranted(String requestId, Authentication authentication, String action, TransportMessage msg,
                               AuthorizationInfo authorizationInfo) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.accessGranted(requestId, authentication, action, msg, authorizationInfo);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.accessGranted(requestId, authentication, action, msg, authorizationInfo);
         }
     }
 
     @Override
     public void accessDenied(String requestId, Authentication authentication, String action, TransportMessage message,
                              AuthorizationInfo authorizationInfo) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.accessDenied(requestId, authentication, action, message, authorizationInfo);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.accessDenied(requestId, authentication, action, message, authorizationInfo);
         }
     }
 
     @Override
     public void tamperedRequest(String requestId, RestRequest request) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.tamperedRequest(requestId, request);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.tamperedRequest(requestId, request);
         }
     }
 
     @Override
     public void tamperedRequest(String requestId, String action, TransportMessage message) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.tamperedRequest(requestId, action, message);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.tamperedRequest(requestId, action, message);
         }
     }
 
     @Override
     public void tamperedRequest(String requestId, User user, String action, TransportMessage request) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.tamperedRequest(requestId, user, action, request);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.tamperedRequest(requestId, user, action, request);
         }
     }
 
     @Override
     public void connectionGranted(InetAddress inetAddress, String profile, SecurityIpFilterRule rule) {
-        if (licenseState.isAuditingAllowed()) {
```

Comment:
The connection methods are called from `IPFilter#accept` so I believe we also need to update the check there